Bridge/src/mqtt.cpp

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include <thread>
+
+#include "mqtt.hpp"
+
+MqttCallback::MqttCallback(mqtt::async_client_ptr client)
+    : client_(client) {}
+
+void MqttCallback::set_on_message(std::function<void(mqtt::const_message_ptr)> callback)
+{
+    this->on_message_ = callback;
+}
+
+void MqttCallback::on_failure(const mqtt::token& asyncActionToken)
+{
+    std::cout << "MQTT connection failed" << std::endl;
+    exit(1);
+}
+
+// Either this or connected() may be used
+void MqttCallback::on_success(const mqtt::token& asyncActionToken) {}
+
+void MqttCallback::connected(const std::string& cause)
+{
+    std::cout << "MQTT connection succeeded" << std::endl;
+    this->client_->subscribe("#", 0);
+}
+
+void MqttCallback::connection_lost(const std::string& cause)
+{
+    std::cout << "MQTT connection lost" << std::endl;
+}
+
+void MqttCallback::message_arrived(mqtt::const_message_ptr message)
+{
+    this->on_message_(message);
+}
+
+
+MqttBridgeClient::MqttBridgeClient()
+{
+    const auto client_id = this->generate_client_id();
+    this->client_ = std::make_shared<mqtt::async_client>("tcp://localhost:1883", client_id);
```

Comment:
Good point, I could pretty easily make it an argument of `MqttBridgeClient`. I'll do that now!

Reason I didn't make it a command line argument is because afaik C++ doesn't have anything like argsparse built in. I imagine I'd get some 3rd party library to handle arguments.

Reply:
Since we only have 1 argument right now, just using `argv[1]` should suffice

Reply:
Very good point! Done that now.

![image](https://user-images.githubusercontent.com/17876556/90096660-351a8a80-dd77-11ea-9cab-cf4c1919469e.png)